clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -351,6 +362,213 @@ class AzureNetworkClient extends AzureBaseClient {
     null
   }
 
+  /**
+   * It creates the server group corresponding backend address pool entry in the selected load balancer
+   *  This will be later used as a parameter in the create server group deployment template
+   * @param resourceGroupName the name of the resource group to look into
+   * @param loadBalancerName the of the application gateway
+   * @param serverGroupName the of the application gateway
+   * @return a resource id for the backend address pool that got created or null/Runtime Exception if something went wrong
+   */
+  String createLoadBalancerAPforServerGroup(String resourceGroupName, String loadBalancerName, String serverGroupName) {
+    def loadBalancer = executeOp({
+      azure.loadBalancers().getByResourceGroup(resourceGroupName, loadBalancerName)
+    })
+
+    if(loadBalancer) {
+      // the application gateway must have an backend address pool list (even if it might be empty)
+      if (!loadBalancer.backends()?.containsKey(serverGroupName)) {
+        String sgTag = loadBalancer.tags().get("serverGroups")
+        sgTag = sgTag == null || sgTag.length() == 0 ? serverGroupName : sgTag + " " + serverGroupName
+        loadBalancer.update()
+          .withTag("serverGroups", sgTag)
+          .defineBackend(serverGroupName)
+          .attach()
+          .apply()
+
+        log.info("Adding backend address pool to ${loadBalancer.name()} for server group ${serverGroupName}")
+      }
+
+      return "${loadBalancer.id()}/backendAddressPools/${serverGroupName}"
+    } else {
+      throw new RuntimeException("Load balancer ${loadBalancerName} not found in resource group ${resourceGroupName}")
+    }
+
+    null
```

Comment:
Please use `return` statements for the groovy methods that aren't simple one-liners.